src/Expr.h

```diff
@@ -226,6 +231,9 @@ class NameExpr final : public Expr {
 
 	TraversalCode Traverse(TraversalCallback* cb) const override;
 
+	// Optimization-related:
```

Comment:
I know this has been discussed in the previous PR, but I'm still confused: is this functionality optimizer-specific or not? I thought the is answer is "no", but the consistent comments still leave it ambiguous.

Reply:
Well, yes & no.  Yes in that it's only used by script optimization.  No in that it doesn't "know" anything about how script optimization works.  The next round of changes are going to add more methods flagged `Optimization-related`, so we'll have that section in the class definitions anyway (modulo resolution of your point below).